packages/lib-runtime/src/store/useResolvedExtensions.ts

```diff
@@ -0,0 +1,80 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import type {
+  Extension,
+  LoadedExtension,
+  ResolvedExtension,
+  ExtensionPredicate,
+} from '../types/extension';
+import { consoleLogger } from '../utils/logger';
+import { settleAllPromises } from '../utils/promise';
+import { isExtensionCodeRefsResolutionError, resolveCodeRefValues } from './coderefs';
+import { usePluginManager } from './PluginStoreContext';
+import { useExtensions } from './useExtensions';
+
+type UseResolvedExtensionsResult<TExtension extends Extension> = [
+  resolvedExtensions: LoadedExtension<ResolvedExtension<TExtension>>[],
+  resolved: boolean,
+  errors: unknown[],
+];
+
+/**
+ * React hook that calls `useExtensions` and resolves all code references in all matching extensions.
+ *
+ * Resolving code references to the corresponding values is an asynchronous operation. Initially,
+ * this hook returns a pending result tuple `[resolvedExtensions: [], resolved: false, errors: []]`.
+ *
+ * Once the resolution is complete, this hook re-renders the component with a result tuple containing
+ * all matching extensions that had their code references resolved successfully along with any errors
+ * that occurred during the process.
+ *
+ * When the list of matching extensions changes, the resolution is restarted.
+ *
+ * The hook's result tuple elements are guaranteed to be referentially stable across re-renders.
```

Comment:
This means SDK users should always destructure the result tuple, i.e.

```ts
const [resolvedExtensions, resolved, errors] = useResolvedExtensions();
```

The result tuple itself is always a new array and therefore not referentially stable across re-renders.